nextjs/components/EmployeeView.tsx

```diff
@@ -1,36 +1,76 @@
 import { useAuth } from '@/context/AuthContext'
-import { FullUserDTO } from '@/types'
+import { FullUserDTO, WorkHourDTO } from '@/types'
 import { env } from 'process'
 import React, { useEffect, useState } from 'react'
 import Calendar from 'react-calendar'
 import 'react-calendar/dist/Calendar.css'
 import CircularButton from './CircularButton'
-
-interface WorkHoursData {
-  [date: string]: number
-}
+import { format } from 'date-fns'
+import styles from '@/styles/employeeview.module.css'
 
 function EmployeeView() {
-  const [date, setDate] = useState<Date>(new Date())
   const { user } = useAuth()
-  const [workHoursData, setWorkHoursData] = useState<WorkHoursData>({})
   const [dbUser, setDbUser] = useState<FullUserDTO | undefined>(undefined)
-  const [selectedUser, setSelectedUser] = useState<FullUserDTO | undefined>()
   useEffect(() => {
     // Fetch the list of users from your database API and update the state
     // Replace the URL with your actual API endpoint
     fetch(`/api/user?username=${user?.preferred_username}`)
       .then((response) => response.json())
       .then((data) => setDbUser(data))
       .catch((error) => console.error(error))
-  }, [])
+  }, [user])
 
+  if (!dbUser) {
+    return
```

Comment:
You might consider having a helpful message here, that the db user was not found.